src/ORM/FieldType/DBCurrency.php

```diff
@@ -62,9 +61,11 @@ public function setValue($value, $record = null, $markChanged = true)
         if (is_numeric($value)) {
             $this->value = $value;
         } elseif (preg_match('/-?\$?[0-9,]+(.[0-9]+)?([Ee][0-9]+)?/', $value, $matches)) {
-            $this->value = str_replace(array('$',',',$this->config()->currency_symbol), '', $matches[0]);
-        } else {
-            $this->value = 0;
+            $this->value = str_replace(['$', ',', $this->config()->currency_symbol], '', $matches[0]);
         }
+
+        $this->value = 0;
```

Comment:
This method doesn't return inside each condition, so this would override every value with zero. If you could put the else clause back this looks like a good change (as `DBField::setValue` returns `$this` so it's a fix rather than an enhancement) 😃

Reply:
Got me